File: 1/1twoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        // iterate throu the array with a nested loop

        for (int i = 1; i < nums.length; i ++) {
            for (int j = i; j < nums.length; j++) {
                // checking if the sum of the current pair equals the target
                if (nums[j] + nums[j - 1] == target) {
                    return new int[] {j, j - i}; // return indicies of the matching pair
                }

            }
        }
        return new int[]{};
    }
}

File: 1/1-twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_map = {} # Dictionarz to store numbers and their indicies

        for i, num in enumerate(nums):
            complement = target - num # The value we need to find

            if complement in num_map:
                return [num_map[complement], i] #Return indicies if complement found
 
            num_map[num] = i # Store the number with its index
        
        return [] # No Solution found
        